rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/SessionsResource.java

```diff
@@ -64,49 +63,125 @@ public Stream<SessionRepresentation> realmSessions(@QueryParam("type") @DefaultV
+                sessionIdStream = Stream.concat(sessionIdStream, clientSessionStats
+                        .keySet().stream().map(i -> new ClientIdSessionType(i, REGULAR)));
+            }
+            if (type == ALL || type == OFFLINE) {
+                sessionIdStream = Stream.concat(sessionIdStream, session.sessions().getActiveClientSessionStats(realm, true)
+                        .keySet().stream().map(i -> new ClientIdSessionType(i, OFFLINE)));
+            }
 
-        Stream<SessionRepresentation> result = sessionIdStream.flatMap((sessionId) -> {
-            ClientModel clientModel = realm.getClientById(sessionId.getClientId());
-            switch (sessionId.getType()) {
-                case REGULAR:
-                    return session.sessions().getUserSessionsStream(realm, clientModel)
-                            .map(s -> toUserSessionRepresentation(s, sessionId.getClientId(), REGULAR));
-                case OFFLINE:
-                    return session.sessions()
-                            .getOfflineUserSessionsStream(realm, clientModel, null, null)
-                            .map(s -> toUserSessionRepresentation(s, sessionId.getClientId(), OFFLINE));
+            Stream<SessionRepresentation> result = sessionIdStream.flatMap((clientIdSessionType) -> {
+                // TODO: Will not work if it is an external client IDs, see JpaUserSessionPersister#getUserSessionsCountsByClients
+                ClientModel clientModel = realm.getClientById(clientIdSessionType.getClientId());
+                switch (clientIdSessionType.getType()) {
+                    case REGULAR:
+                        return session.sessions().getUserSessionsStream(realm, clientModel)
+                                .map(s -> toRepresentation(s, REGULAR));
+                    case OFFLINE:
+                        return session.sessions()
+                                .getOfflineUserSessionsStream(realm, clientModel, null, null)
+                                .map(s -> toRepresentation(s, OFFLINE));
+                }
+                return Stream.<SessionRepresentation>builder().build();
+            }).distinct();
+
+            if (!search.equals("")) {
+                result = result.filter(s -> s.getUsername().contains(search) || s.getIpAddress().contains(search)
+                                            || s.getClients().values().stream().anyMatch(c -> c.contains(search)));
+            }
+            return result.skip(first).limit(max);
+        } else {
+            ClientModel client = null;
+            UserModel user = null;
+            String ipAddress;
+            if (!search.isEmpty()) {
+                user = session.users().getUserByUsername(realm, search);
+                if (user != null) {
+                    if (!auth.users().canView(user)) {
+                        return Stream.empty();
```

Comment:
Should this be returning an empty stream or just ignoring the user search? Suppose a search with `broker` string; existing user `broker` would be sufficient to return an empty stream while there could be numerous `broker` client sessions.

Reply:
this has been reworked, I'm sticking with the original logic